### PR TITLE
Project History

### PR DESCRIPTION
**Pull request by csansoon**
I feel like we're making history here.

https://github.com/user-attachments/assets/673309df-57c6-4612-8543-476e1007470e

### CODE REVIEW

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx [hunk 1]
```diff
@@ -309,7 +311,39 @@ export default function DocumentEditor({
...
+                            <Button
+                              variant='outline'
+                              size='small'
+                              className='h-7'
+                            >
+                              <Icon
+                                name='history'
+                                color='foregroundMuted'
+                                className='h-4 w-4'
+                              />
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx:337]: Button has `iconProps` 🙏 no need for Icon inside button

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx [hunk 2]
```diff
@@ -309,7 +311,39 @@ export default function DocumentEditor({
...
+                                .commits.detail({ uuid: commit.uuid })
+                                .history.detail({
+                                  uuid: document.documentUuid,
+                                }).root
+                            }
+                          >
+                            <Button
+                              variant='outline'
+                              size='small'
+                              className='h-7'
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx:331]: Why hardcoded height?

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/history/_components/CommitChangesList/index.tsx [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            <TruncatedTooltip content={change.path} className={dimmedClass}>
+              <Text.H5M color={color} ellipsis noWrap>
+                {change.path}
+              </Text.H5M>
+            </TruncatedTooltip>
+          </div>
+          <Icon
+            name={icon}
+            className={cn(
+              'flex-shrink-0 w-4 h-4',
```
Comment (andresgutgon) [inline apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/history/_components/CommitChangesList/index.tsx:121]: `w-4` is default size of icon

File: packages/web-ui/src/ds/molecules/TruncatedTooltip/index.tsx [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        open={isOverflowing ? undefined : false}
+        trigger={
+          <div ref={textRef} className={cn('max-w-full truncate', className)}>
+            {children}
+          </div>
+        }
+      >
+        {content}
+      </Tooltip>
+    </div>
```
[the remark this answers is not in the document]
Comment (csansoon) [reply packages/web-ui/src/ds/molecules/TruncatedTooltip/index.tsx:50]: I made this component for wrapping single-line other Text components.

If you wrap a `<Text.X noWrap ellipsis>...` inside this component, then a tooltip will appear when hovering over the text only if it has been truncated :D

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx (hunk 2 above)
Comment (csansoon) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx:331]: image

Comment (andresgutgon) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx:331]: Why h-7? Is not part of our design system no?

Comment (csansoon) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx:330]: For some reason the `DropdownMenuTrigger` has `5px` of vertical padding (?)

File: packages/web-ui/src/ds/molecules/TruncatedTooltip/index.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (csansoon) [reply packages/web-ui/src/ds/molecules/TruncatedTooltip/index.tsx:50]: https://github.com/user-attachments/assets/e6c6f44b-50d1-46f7-8738-6569ad48a30a

looks cool :)

[the remark this answers is not in the document]
Comment (andresgutgon) [reply packages/web-ui/src/ds/molecules/TruncatedTooltip/index.tsx:50]: Very nice!

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx (hunk 2 above)
Comment (andresgutgon) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx:330]: I don't like this tbh

**Review by andresgutgon — DISMISSED** [review]

File: apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx (hunk 2 above)
Comment (csansoon) [reply apps/web/src/app/(private)/projects/[projectId]/versions/[commitUuid]/documents/[documentUuid]/_components/DocumentEditor/Editor/index.tsx:330]: I can't exactly point where is this padding being added

File: apps/web/src/stores/commitChanges.ts [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  const fetcher = useFetcher(route)
+
+  const { data = [], ...rest } = useSWR<ChangedDocument[]>(
+    ['commitChanges', commit?.projectId, commit?.uuid],
+    fetcher,
+    opts,
+  )
+
+  return {
+    data: data ?? [],
```
Comment (geclos) [inline apps/web/src/stores/commitChanges.ts:25]: useFetcher has a fallback optional parameter for this kind of defaults (and in fact i think it already defaults to empty array

File: packages/core/src/repositories/commitsRepository/index.ts [hunk 6]
```diff
@@ -129,6 +127,55 @@ export class CommitsRepository extends RepositoryLegacy<
...
+          eq(this.scope.projectId, commit.projectId),
+          mergedAtFilter,
+          not(eq(this.scope.id, commit.id)),
+        ),
+      )
+      .orderBy(desc(this.scope.mergedAt))
+      .limit(1)
+      .then((r) => r[0])
+  }
+
```
Comment (geclos) [inline packages/core/src/repositories/commitsRepository/index.ts:178]: good opportunity for a test here

**Review by geclos — COMMENTED** [review]
lovely although a bit worriseome that a 1k line pr has 0 new tests.

Btw in the video you shared it looks like the performance loading the history is quite bad. Is that because of dev mode or is there an underlying issue?

**Pull request merged**